src/main/java/io/aiven/kafka/connect/s3/config/AwsCredentialProviderFactory.java

```diff
@@ -58,6 +58,9 @@ private AWSSecurityTokenService securityTokenService(final S3SinkConfig config)
 
     private AWSCredentialsProvider getBasicAwsCredentialsProvider(final S3SinkConfig config) {
```

Comment:
As the method doesn't only return a `BasicAWSCredentials` anymore, but conditionally some custom implementation, let's move logic from private method into `getProvider` where it's called to make this clear.

Reply:
Updated as per suggestion.